src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseAsyncMethodInAsyncContext.cs

```diff
@@ -148,6 +165,21 @@ public override void Initialize(AnalysisContext context)
             });
         }
 
+        private static bool IsSemaphoreSlimWaitWithZeroArgumentInvocation(IInvocationOperation invocation, IFieldSymbol? timeSpanZero, ISet<IMethodSymbol> semaphoreSlimWaitMethods)
+        {
+            if (!semaphoreSlimWaitMethods.Contains(invocation.TargetMethod, SymbolEqualityComparer.Default) || invocation.Arguments.IsEmpty)
```

Comment:
The arguments length check here is redundant, right? I would expect IOperation invocation to always have the same number of arguments as parameters, and we already know from the set creation above that only overloads with one or more parameters is contained in the collection.

```suggestion
            if (!semaphoreSlimWaitMethods.Contains(invocation.TargetMethod, SymbolEqualityComparer.Default))
```

If you remove the check, feel free to include a `Debug.Assert` line to make it clear for future readers users why it was excluded.